problems/http.go

```diff
@@ -31,10 +31,13 @@ func WriteResponse(ctx context.Context, err error, w http.ResponseWriter, r *htt
 		WithError(err).
 		WithField("code", statusCode)
 
-	// Log as an Error if statusCode is 5XX, otherwise as Info.
-	if statusCode/100 == http.StatusInternalServerError/100 {
+	// Log as an Error if statusCode is 5XX, Warn if context was cancelled, otherwise Info.
+	switch {
+	case statusCode/100 == http.StatusInternalServerError/100:
```

Comment:
All errors (including canceled ones) will be converted to `InternalProblem`, which have the status code of `500`. The case statement below will never be run?

Adding a few unit tests here might be good?